test/true-currencies/TrueCurrencyWithPoR.test.ts

```diff
@@ -0,0 +1,184 @@
+
+    // Deploy a mock aggregator to mock PoR feed answers
+    mockV3Aggregator = await new MockV3Aggregator__factory(owner).deploy(
+      '18',
+      TUSD_FEED_INITIAL_ANSWER,
+    )
+  })
+
+  beforeEach(async () => {
+    // Reset pool PoR feed defaults
+    const currentFeed = await token.chainReserveFeed()
+    if (currentFeed.toLowerCase() !== mockV3Aggregator.address.toLowerCase()) {
+      await token.setChainReserveFeed(mockV3Aggregator.address)
+    }
+    const currentHeartbeat = await token.chainReserveHeartbeat()
+    const MAX_AGE = await token.MAX_AGE()
+    if (!currentHeartbeat.eq(MAX_AGE)) {
+      await token.setChainReserveHeartbeat(0)
+    }
+
+    // Set fresh, valid answer on mock PoR feed
+    const tusdSupply = await token.totalSupply()
+    await mockV3Aggregator.updateAnswer(tusdSupply.add(AMOUNT_TO_MINT))
+  })
+
+  it('should mint successfully when feed is unset', async () => {
+    // Make sure feed is unset
+    await token.setChainReserveFeed(AddressZero)
+    expect(await token.chainReserveFeed()).to.equal(AddressZero)
+
+    // Mint TUSD
+    const balanceBefore = await token.balanceOf(owner.address)
+    await token.mint(owner.address, AMOUNT_TO_MINT)
+    expect(await token.balanceOf(owner.address)).to.equal(balanceBefore.add(AMOUNT_TO_MINT))
+  })
+
+  it('should mint successfully when feed is set, but heartbeat is unset (defaulting to MAX_AGE)', async () => {
+    // Mint TUSD
+    const balanceBefore = await token.balanceOf(owner.address)
+    await token.mint(owner.address, AMOUNT_TO_MINT, {
+      gasLimit: 200_000,
+    })
+    expect(await token.balanceOf(owner.address)).to.equal(AMOUNT_TO_MINT.add(balanceBefore))
+  })
+
+  it('should mint successfully when both feed and heartbeat are set', async () => {
+    // Set heartbeat to 1 day
+    await token.setChainReserveHeartbeat(ONE_DAY_SECONDS)
+    expect(await token.chainReserveHeartbeat()).to.equal(ONE_DAY_SECONDS)
+
+    // Mint TUSD
+    const balanceBefore = await token.balanceOf(owner.address)
+    await token.mint(owner.address, AMOUNT_TO_MINT)
+    expect(await token.balanceOf(owner.address)).to.equal(balanceBefore.add(AMOUNT_TO_MINT))
+  })
+
+  it('should mint successfully when feed decimals < TrueCurrency decimals', async () => {
+    // Re-deploy a mock aggregator with fewer decimals
+    const currentTusdSupply = await token.totalSupply()
+    const validReserve = currentTusdSupply.div(exp(1, 12)).add(AMOUNT_TO_MINT)
+    const mockV3AggregatorWith6Decimals = await new MockV3Aggregator__factory(owner).deploy('6', validReserve)
+    // Set feed and heartbeat on newly-deployed aggregator
+    await token.setChainReserveFeed(mockV3AggregatorWith6Decimals.address)
+    expect(await token.chainReserveFeed()).to.equal(mockV3AggregatorWith6Decimals.address)
+    await token.setChainReserveHeartbeat(ONE_DAY_SECONDS)
+    expect(await token.chainReserveHeartbeat()).to.equal(ONE_DAY_SECONDS)
```

Comment:
We don't need to verify every time that setting the heartbeat works. It's enough to verify it in one test.

Reply:
Removed redundant checks.